src/Raven.Server/Commercial/LetsEncrypt/LetsEncryptCertificateUtil.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Org.BouncyCastle.Asn1;
+using Org.BouncyCastle.Asn1.X509;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Pkcs;
+using Raven.Client.ServerWide.Operations.Certificates;
+using Raven.Server.Utils;
+using Sparrow.Server.Platform.Posix;
+
+namespace Raven.Server.Commercial.LetsEncrypt;
+
+public class LetsEncryptCertificateUtil
+{
+    public static string GetServerUrlFromCertificate(X509Certificate2 cert, SetupInfo setupInfo, string nodeTag, int port, int tcpPort, out string publicTcpUrl, out string domain)
```

Comment:
Is there anything specific to Let's Encrypt cert here? If not and it's more generic then maybe it could be part of `CertificateUtils`?

Reply:
The main usage is within the `zipFileHelper`, Yes, it's better to move it to a more generic place.

Reply:
done